the one time we refactored all the form things

sturobson opened the pull request:
issue here - https://github.com/visual-framework/vf-core/issues/820

sturobson committed:
removes floatlabel from forms

sturobson committed:
whitepsace

khawkins98 commented on components/vf-form/vf-form__core/assets/vf-form__float-labels.js:
```diff
@@ -1,127 +0,0 @@
-// vf-form__float-labels
```
Rather than delete all the JS, it would be better if we left the function with a console.log or .warn, noting that your need to update your scripts.js rollup. That way a component update won't break the site's JS and alert the dev they need to update.

sturobson replied:
added to this PR now

sturobson committed:
layout of form inputs and labels

Updates to the layout of form inputs and labels as well as the radio and checkbox layouts

sturobson committed:
reinstates JS with console log

sturobson committed:
reinstates JS with console log

sturobson committed:
refactor example page

we have moved the examples to their own --example.njk file as is now best practice

sturobson committed:
adds fieldset and legend

adds vf-form__fieldset and vf-form__legend components

sturobson committed:
tidies up Sass for label

replaces hard coded margin bottom to using Sass map of spacing

sturobson committed:
tidies vf-form__label

this removes the input from the label .njk file. It also adds a 'required' label with a red astericks

sturobson committed:
tidies vf-form__label

this removes the input from the label .njk file. It also adds a 'required' label with a red astericks

sturobson committed:
adds form error helper

sturobson committed:
adds full-feature form item

adds input, label, required label, helper text, and error text inside the

khawkins98 commented on components/vf-form/vf-form__fieldset/README.md:
```diff
@@ -0,0 +1,59 @@
+👋 Hi component authour, you've read the docs, right?
```
These first 12 lines of boilerplate got left in. I should probably add a note to delete this.

sturobson replied:
or it's a pre-commit to check if it's there and ask to delete it in the cli?

sturobson replied:
because search has the action in the button text - I think we can do without having the label visible. 

I've snuck in a slight CSS change for the inline version so the button lines up nicely.

khawkins98 commented on components/vf-form/vf-form__helper/vf-form__helper.scss:
```diff
@@ -13,10 +13,10 @@
-    position: relative;
-    top: 12px;
-  }
+.vf-form__helper--error {
```
Should we add any CSS for form element `:invalid` states ... which I guess would be turning the border red or something? (perhaps similar to `vf-form__input:disabled`)

sturobson replied:
good idea - will add

khawkins98 reviewed (commented):
A couple of other misc:

- we should get the changleogs updated (happy to help)
- vf-search also needs updating as it has `vf-form__label vf-u-sr-only | vf-search__label`

sturobson committed:
adds invalid/required style

adds border to input when a field is required but empty. The example also includes the required label asterick and the error text that would be required too

sturobson commented:
added required/`:invliad` styles and example

![Screenshot 2020-03-31 at 14 09 37](https://user-images.githubusercontent.com/925197/78029997-58f5aa80-7359-11ea-9199-8dfe8d55926b.png)

sturobson committed:
tidies up search CSS

removes CSS that wasn't working, positioned the button for inline searches

sturobson committed:
adds inline checkbox example

sturobson committed:
adds all th changelogs

sturobson committed:
adds all th changelogs

khawkins98 committed:
Some cleanup for the vf-forms PR

Spotted a couple things in #821, figured a PR would be faster than explaining.

- sets the human names for the new components
- hides the new components from the overview
- removes some boilerplate
- sneaks in typo fix for vf-lede

khawkins98 committed:
Remove superflous changelog info

There's a lot more of these to cleanup. But it's a start

khawkins98 committed:
Merge pull request #823 from visual-framework/enhancement/cleanup-vf-forms

Some cleanup for the vf-forms PR

sturobson committed:
form documentation

starts adding some simple form documentation

sturobson commented:
I've merged this in now so it's ready to be used after our meeting with EBI nG developers today.
